src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/TextEditingTarget.java

```diff
@@ -6303,6 +6298,44 @@ public TextEditingTargetNotebook getNotebook()
       return notebook_;
    }
    
+   private void setRMarkdownBehaviorEnabled(boolean enabled)
+   {
+      // register idle monitor; automatically creates/refreshes previews
+      // of images and LaTeX equations during idle
+      if (bgIdleMonitor_ == null && enabled)
+         bgIdleMonitor_ = new TextEditingTargetIdleMonitor(this, 
+               docUpdateSentinel_);
+      else if (bgIdleMonitor_ != null)
+      {
+         if (enabled)
+            bgIdleMonitor_.beginMonitoring();
+         else
+            bgIdleMonitor_.endMonitoring();
+      }
+      
+      // set up mathjax
+      if (mathjax_ == null && enabled)
+         mathjax_ = new MathJax(docDisplay_, docUpdateSentinel_, prefs_);
+
+      if (enabled)
+      {
+         // auto preview images and equations
+         new ImagePreviewer(docDisplay_, docUpdateSentinel_, prefs_);
```

Comment:
Do we need to keep track of / detach this object when the associated document is closed? It looks like we register some property changed handlers on the doc update sentinel; do those automatically get de-registered when the document is closed?